ixmp/backend/jdbc.py

```diff
@@ -802,15 +857,31 @@ def unwrap(v):
         return {entry.getKey(): unwrap(entry.getValue())
                 for entry in self.jindex[s].getMeta().entrySet()}
 
-    def set_meta(self, s, name, value):
+    def set_meta(self, s, name_or_data, value=None):
+        if type(name_or_data) == dict:
+            name_or_data = list(name_or_data.items())
```

Comment:
These two lines only seem like something that would be done to any user input, regardless of which Backend is in use. Is that accurate? If that's so, then it should be done in core.py.